src/hotspot/cpu/x86/x86.ad

```diff
@@ -2111,13 +2136,23 @@ bool Matcher::match_rule_supported_vector_masked(int opcode, int vlen, BasicType
 
     case Op_MaxV:
     case Op_MinV:
+    case Op_UMinV:
+    case Op_UMaxV:
       if (is_subword_type(bt) && !VM_Version::supports_avx512bw()) {
         return false; // Implementation limitation
       }
       if (is_floating_point_type(bt)) {
         return false; // Implementation limitation
       }
       return true;
+    case Op_SaturatingAddVB:
+    case Op_SaturatingAddVS:
+    case Op_SaturatingSubVB:
+    case Op_SaturatingSubVS:
+      if (size_in_bits < 128 || !VM_Version::supports_avx512bw()) {
+        return false; // Implementation limitation
+      }
+      return true;
```

Comment:
Byte/Short saturating vector add is supported for AVX1, AVX2 platforms. Could we not do register version for size_in_bit < 128?